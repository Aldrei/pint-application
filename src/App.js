import React from 'react'

import logo from './logo.svg'
import './App.css'

import Stack from '@mui/material/Stack'

import Button from './components/Button'
import {Counter} from './features/counter/Counter'

import {
    BrowserRouter as Router,
    Routes,
    Route,
    Link
} from 'react-router-dom'

function App() {
    return (
        <div className="App">
            <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <h4>Material UI Test</h4>
                <Stack spacing={2} direction="row">
                    <Button variant="text">Text</Button>
                    <Button variant="contained">Contained</Button>
                    <Button variant="outlined">Outlined</Button>
                </Stack>
            </header>
            <Router>
                <div>
                    <nav>
                        <ul>
                            <li>
                                <Link to="/">Home</Link>
                            </li>
                            <li>
                                <Link to="/counter">Counter</Link>
                            </li>
                        </ul>
                    </nav>
                    <Routes>
                        <Route path="/counter" element={<Counter />} />
                    </Routes>
                </div>
            </Router>
        </div>
    )
}

export default App
