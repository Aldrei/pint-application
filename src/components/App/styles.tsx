import styled from 'styled-components';

export const AppContainer = styled.div`
  width: 100%;
  overflow-x: hidden;
  border: 2px solid green;
`;
