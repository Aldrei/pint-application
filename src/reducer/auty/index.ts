import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { authService, IAuthServiceAccessTokenRequest, IAuthServiceAccessTokenResponse } from '../../services/auth';

import { IServiceRequest } from '../../types';

export interface IAutyState extends IServiceRequest {
  whoIsAuth: object;
  accessToken: IAuthServiceAccessTokenResponse;
}

const initialState: IAutyState = {
  status: 'idle',
  whoIsAuth: {},
  accessToken: {} as IAuthServiceAccessTokenResponse,
};

// The function below is called a thunk and allows us to perform async logic. It
// can be dispatched like a regular action: `dispatch(incrementAsync(10))`. This
// will call the thunk with the `dispatch` function as the first argument. Async
// code can then be executed and other actions can be dispatched. Thunks are
// typically used to make async requests.
export const authServiceThunk = createAsyncThunk(
  'counter/fetchCount',
  async (data: IAuthServiceAccessTokenRequest) => {
    const response = await authService.accessToken(data);
    // The value we return becomes the `fulfilled` action payload
    return response;
  }
);

export const authSlice = createSlice({
  name: 'auth',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {},
  // The `extraReducers` field lets the slice handle actions defined elsewhere,
  // including actions generated by createAsyncThunk or in other slices.
  extraReducers: (builder) => {
    builder
      .addCase(authServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(authServiceThunk.fulfilled, (state, action) => {
        console.log('authServiceThunk fulfilled:', action);
        state.status = 'success';
        state.accessToken = action.payload?.data?.access_token ? action.payload.data : {};
        state.data = action.payload.data;
      })
      .addCase(authServiceThunk.rejected, (state, action) => {
        console.log('authServiceThunk rejected:', action);
        state.status = 'failed';
        state.data = action.payload as object;
        state.accessToken = {};
      });
  },
});

// The function below is called a selector and allows us to select a value from
// the state. Selectors can also be defined inline where they're used instead of
// in the slice file. For example: `useSelector((state: RootState) => state.counter.value)`
// export const selectCount = (state: RootState) => state.counter.value;

export default authSlice.reducer;
