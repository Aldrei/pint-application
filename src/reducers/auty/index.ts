import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { authService, IAuthServiceAccessTokenRequest, IAuthServiceAccessTokenResponse, IAuthServiceRevokeRequest } from '../../services/auth';
import { AppThunk } from '../../stores';

import { IServiceRequest, IWhoIsAuth } from '../../types';

export interface IAutyState extends IServiceRequest {
  whoIsAuth: IWhoIsAuth;
  accessToken: IAuthServiceAccessTokenResponse;
}

const initialState: IAutyState = {
  name: 'authReducer',
  status: 'idle',
  whoIsAuth: {} as IWhoIsAuth,
  accessToken: {} as IAuthServiceAccessTokenResponse,
};

export const authServiceThunk = createAsyncThunk(
  'auth/access_token',
  async (data: IAuthServiceAccessTokenRequest) => {
    const response = await authService.accessToken(data);
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const revokeServiceThunk = createAsyncThunk(
  'auth/revoke',
  async (data: IAuthServiceRevokeRequest) => {
    const response = await authService.revoke(data);
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const whoServiceThunk = createAsyncThunk(
  'auth/who_is_auth',
  async () => {
    const response = await authService.who();
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const authSlice = createSlice({
  name: 'auth',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {
    clearAccessToken: (state) => {
      state.accessToken = {} as IAuthServiceAccessTokenResponse;
      state.whoIsAuth = {} as IWhoIsAuth;
    }
  },
  // The `extraReducers` field lets the slice handle actions defined elsewhere,
  // including actions generated by createAsyncThunk or in other slices.
  extraReducers: (builder) => {
    builder
      /** Access token */
      .addCase(authServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(authServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.accessToken = action.payload?.data?.access_token ? action.payload.data : {};
        state.data = action.payload.data;
      })
      .addCase(authServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.data = action.payload as object;
        state.accessToken = {};
      })
      /** Revoke. */
      .addCase(revokeServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(revokeServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.data = action.payload.data;
        state.accessToken = {};
        state.whoIsAuth = {} as IWhoIsAuth;
      })
      .addCase(revokeServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.data = action.payload as object;
      })
      /** Who is auth. */
      .addCase(whoServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(whoServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.whoIsAuth = action.payload.data;
      })
      .addCase(whoServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.whoIsAuth = action.payload as IWhoIsAuth;
      });
  },
});

const { clearAccessToken } = authSlice.actions;

export const clearAccessTokenReducer = (): AppThunk => (dispatch) => {
  dispatch(clearAccessToken());
};

export default authSlice.reducer;
