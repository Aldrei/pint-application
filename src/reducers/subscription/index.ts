import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { ISubscriptionPaymentResponse, ISubscriptionPaymentServiceRequest, subscriptionService } from '../../services/subscription';
import { AppThunk } from '../../stores';

import { IServiceRequest } from '../../types';

export interface ISubscriptionState extends IServiceRequest {
  payment: ISubscriptionPaymentResponse;
}

const initialState: ISubscriptionState = {
  name: 'subscriptionReducer',
  status: 'idle',
  payment: {} as ISubscriptionPaymentResponse,
};

export const subscriptionPaymentServiceThunk = createAsyncThunk(
  'subscription/payment',
  async (data: ISubscriptionPaymentServiceRequest) => {
    const response = await subscriptionService.payment(data);
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const subscriptionSlice = createSlice({
  name: 'auth',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {
    clearAccessToken: (state) => {
      state.payment = {} as ISubscriptionPaymentResponse;
    }
  },
  // The `extraReducers` field lets the slice handle actions defined elsewhere,
  // including actions generated by createAsyncThunk or in other slices.
  extraReducers: (builder) => {
    builder
      /** Payment */
      .addCase(subscriptionPaymentServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(subscriptionPaymentServiceThunk.fulfilled, (state, action) => {
        console.log('DEBUG action.payload:', action.payload);
        state.status = 'success';
        state.data = action.payload.data;
      })
      .addCase(subscriptionPaymentServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.data = action.payload as object;
        state.payment = {};
      });
  },
});

const { clearAccessToken } = subscriptionSlice.actions;

export const clearAccessTokenReducer = (): AppThunk => (dispatch) => {
  dispatch(clearAccessToken());
};

export default subscriptionSlice.reducer;
