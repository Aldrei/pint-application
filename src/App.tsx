import React from 'react';
// import logo from './logo.svg';
import './App.css';

import LoginPage from './pages/Login';

/**
 * Test Slack Notification workflow:
 * 
 *  Test deploy on DEVELOPMENT branch: Test 2
 *  Test deploy on MASTER branch: Test 3
*/

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <LoginPage />
      </header>
    </div>
  );
}

export default App;
