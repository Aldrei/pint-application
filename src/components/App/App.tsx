import React from 'react';

import LoginPage from '../../pages/Login';

import { AppContainer } from './styles';

function App() {
  return (  
    <AppContainer className='appComp'>
      <LoginPage />
    </AppContainer>
  );
}

export default App;
