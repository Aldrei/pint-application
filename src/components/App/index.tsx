import React from 'react';

import { ThemeProvider } from '@mui/material/styles';

import {
  BrowserRouter,
  Routes,
  Route,
  Navigate,
} from 'react-router-dom';

import LoginPage from '../../pages/Login';
import DashboardPage from '../../pages/Dashboard';
import PropertiesList from '../../pages/Properties/List';
import PropertiesDetail from '../../pages/Properties/Detail';
import PropertyCreateEdit from '../../pages/Properties/CreateEdit';
import OwnerCrud from '../../pages/Owners/Crud';
import OwnersList from '../../pages/Owners/List';
import EmployeesList from '../../pages/Employees/List';
import EmployeesCrud from '../../pages/Employees/Crud';
import CitiesList from '../../pages/Cities/List';
import CitiesCrud from '../../pages/Cities/Crud';
import NeighborhoodsList from '../../pages/Neighborhoods/List';
import NeighborhoodsCrud from '../../pages/Neighborhoods/Crud';
import SlidesCreateEdit from '../../pages/Slides/Crud';
import BannersCreateEdit from '../../pages/Banners/Crud/CreateEdit';

import CheckAuth from '../../components/CheckAuth';
import CheckUnauthenticated from '../../components/CheckUnauthenticated';

import { ROUTES } from '../../constants/routes';

import ColorModeContext from '../../contexts/ColorModeContext';
import useTheme from '../../hooks/useTheme';

import SnackContext from '../../contexts/SnackContext';
import useSnack from '../../hooks/useSnack';

import { AppContainer, AppErrorContainer } from './styles';

function App() {
  const { theme, colorMode } = useTheme();
  const { addSnack, SnackList } = useSnack();

  const checkEnvs = () => {
    if (!process.env.REACT_APP_ENVIRONMENT) return false;
    if (!process.env.REACT_APP_API_BASE_URL) return false;
    return true;
  };

  if (!checkEnvs())
    return (
      <AppErrorContainer data-testid='appErrorContainer'>Check your ENVs.</AppErrorContainer>
    );

  return (
    <ColorModeContext.Provider value={colorMode}>
      <ThemeProvider theme={theme}>
        <SnackContext.Provider value={addSnack}>
          {SnackList && <SnackList />}

          <BrowserRouter basename='/'>
            <AppContainer className='appComp' data-testid='appContainer'>
              {/* A <Switch> looks through its children <Route>s and
                renders the first one that matches the current URL. */}
              <Routes>
                <Route path={ROUTES.index.path} element={<Navigate to='/login' />} />
                <Route path={ROUTES.login.path} element={
                  <CheckUnauthenticated>
                    <LoginPage />
                  </CheckUnauthenticated>
                } />
                <Route path={ROUTES.dashboard.path} element={
                  <CheckAuth>
                    <DashboardPage />
                  </CheckAuth>
                } />
                <Route path={ROUTES.propertiesList.path} element={
                  <CheckAuth>
                    <PropertiesList />
                  </CheckAuth>
                } />
                {<Route path={ROUTES.propertiesDetail.path} element={
                  <CheckAuth>
                    <PropertiesDetail />
                  </CheckAuth>
                } />}
                {<Route path={ROUTES.propertiesCreate.path} element={
                  <CheckAuth>
                    <PropertyCreateEdit />
                  </CheckAuth>
                } />}
                {<Route path={ROUTES.propertiesEdit.path} element={
                  <CheckAuth>
                    <PropertyCreateEdit />
                  </CheckAuth>
                } />}
                {/**
                 * Owners 
                 */}
                <Route path={ROUTES.ownersList.path} element={
                  <CheckAuth>
                    <OwnersList />
                  </CheckAuth>
                } />
                {<Route path={ROUTES.ownersCreate.path} element={
                  <CheckAuth>
                    <OwnerCrud action='create' />
                  </CheckAuth>
                } />}
                {<Route path={ROUTES.ownersEdit.path} element={
                  <CheckAuth>
                    <OwnerCrud action='edit' />
                  </CheckAuth>
                } />}
                {<Route path={ROUTES.ownersDetail.path} element={
                  <CheckAuth>
                    <OwnerCrud action='show' />
                  </CheckAuth>
                } />}
                {<Route path={ROUTES.ownersDelete.path} element={
                  <CheckAuth>
                    <OwnerCrud action="delete" />
                  </CheckAuth>
                } />}
                {/** 
                 * Employees 
                 */}
                <Route path={ROUTES.employeesList.path} element={
                  <CheckAuth>
                    <EmployeesList />
                  </CheckAuth>
                } />
                <Route path={ROUTES.employeesCreate.path} element={
                  <CheckAuth>
                    <EmployeesCrud action="create" />
                  </CheckAuth>
                } />
                <Route path={ROUTES.employeesEdit.path} element={
                  <CheckAuth>
                    <EmployeesCrud action="edit" />
                  </CheckAuth>
                } />
                <Route path={ROUTES.employeesDetail.path} element={
                  <CheckAuth>
                    <EmployeesCrud action="show" />
                  </CheckAuth>
                } />
                <Route path={ROUTES.employeesDelete.path} element={
                  <CheckAuth>
                    <EmployeesCrud action="delete" />
                  </CheckAuth>
                } />
                {/** 
                 * Cities 
                 */}
                <Route path={ROUTES.citiesList.path} element={
                  <CheckAuth>
                    <CitiesList />
                  </CheckAuth>
                } />
                <Route path={ROUTES.citiesEdit.path} element={
                  <CheckAuth>
                    <CitiesCrud action="edit" />
                  </CheckAuth>
                } />
                {/** 
                 * Neighborhoods 
                 */}
                <Route path={ROUTES.neighborhoodsList.path} element={
                  <CheckAuth>
                    <NeighborhoodsList />
                  </CheckAuth>
                } />
                <Route path={ROUTES.neighborhoodsEdit.path} element={
                  <CheckAuth>
                    <NeighborhoodsCrud action="edit" />
                  </CheckAuth>
                } />
                {/** 
                 * Slides 
                 */}
                <Route path={ROUTES.slidesEdit.path} element={
                  <CheckAuth>
                    <SlidesCreateEdit />
                  </CheckAuth>
                } />
                <Route path={ROUTES.bannersCreate.path} element={
                  <CheckAuth>
                    <BannersCreateEdit />
                  </CheckAuth>
                } />
              </Routes>
            </AppContainer>
          </BrowserRouter>
        </SnackContext.Provider>
      </ThemeProvider>
    </ColorModeContext.Provider>
  );
}

export default App;
