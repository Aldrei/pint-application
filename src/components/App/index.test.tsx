import * as React from 'react';

import { render } from '@testing-library/react';

import { IAutyState } from '../../reducers/auty';

jest.mock('../../hooks/useReducerSelector', () => {
  const useAppSelectorBlaBlaBalMocked = jest.fn(() => ({
    accessToken: 'this-is-a-invalid-token',
    whoIsAuth: {}
  } as unknown as IAutyState));
  return { useAppSelectorBlaBlaBal: useAppSelectorBlaBlaBalMocked };
});

jest.mock('../../hooks/useReducerDispatch', () => {
  const useAppDispatchMocked = jest.fn(() => null);
  return { useAppDispatch: useAppDispatchMocked };
});

import App from './index';

describe('App component', () => {
  beforeAll(() => {
    process.env = Object.assign(process.env, {
      REACT_APP_ENVIRONMENT: 'local',
      REACT_APP_API_BASE_URL: 'https://blablabla.bla/api'
    });
  });

  it('Should render Login page correctly', () => {
    process.env = Object.assign(process.env, {
      REACT_APP_ENVIRONMENT: 'local',
      REACT_APP_API_BASE_URL: 'https://blablabla.bla/api'
    });

    const bodyEl = render(<App />);

    const appNode = bodyEl.getByTestId('appContainer');
    const loginPage = bodyEl.getByTestId('loginContainer');
    expect(appNode).toContainElement(loginPage);
    
    expect(bodyEl.baseElement).toMatchSnapshot();
  });

  it('There is no REACT_APP_API_BASE_URL env', () => {
    delete process.env.REACT_APP_API_BASE_URL;

    const bodyEl = render(<App />);

    const appNode = bodyEl.getByTestId('appErrorContainer');
    expect(appNode).toHaveTextContent('Check your ENVs.');
    
    expect(bodyEl.baseElement).toMatchSnapshot();
  });

  it('There is no REACT_APP_ENVIRONMENT env', () => {
    delete process.env.REACT_APP_ENVIRONMENT;

    const bodyEl = render(<App />);

    const appNode = bodyEl.getByTestId('appErrorContainer');
    expect(appNode).toHaveTextContent('Check your ENVs.');
    
    expect(bodyEl.baseElement).toMatchSnapshot();
  });
});
