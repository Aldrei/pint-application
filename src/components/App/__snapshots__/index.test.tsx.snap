// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render correctly 1`] = `
<body>
  <div>
    <div
      class="sc-eCYdqJ gvCfOl appComp"
      data-testid="appContainer"
    >
      <div
        class="sc-hKMtZM dkGWvO"
        data-testid="loginContainer"
      >
        <div
          class="sc-gsnTZi dfpLxU cardComp"
        >
          <input
            class="sc-dkzDqf jpsCuT inputComp"
            data-testid="username"
            name="username"
            placeholder="Email"
          />
          <input
            class="sc-dkzDqf jpsCuT inputComp"
            data-testid="password"
            name="password"
            placeholder="Senha"
            type="password"
          />
          <button
            class="sc-bczRLJ daPOd buttonComp"
            color="blue"
            data-testid="button-login"
            disabled=""
          >
            Entrar
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
